dvc/dependency/repo.py

```diff
@@ -46,14 +48,86 @@ def _make_repo(self, **overrides):
         with external_repo(**merge(self.def_repo, overrides)) as repo:
             yield repo
 
+    def _get_checksum_in_repo(self, repo):
+        return repo.cache.local.get_checksum(
+            PathInfo(os.path.join(repo.root_dir, self.def_path))
+        )
+
+    # fileinfo is a dictionary containing "checksum" and "path"
+    def _get_fileinfo(self, updated=False):
+        rev_lock = None
+        if not updated:
+            rev_lock = self.def_repo.get(self.PARAM_REV_LOCK)
+
+        try:
+            with self._make_repo(rev_lock=rev_lock) as repo:
+                path = os.path.join(repo.root_dir, self.def_path)
+                try:
+                    checksum = repo.find_out_by_relpath(self.def_path).info[
+                        "md5"
+                    ]
+                except OutputNotFoundError:
+                    repo.state.load()
+                    checksum = repo.cache.local.get_checksum(PathInfo(path))
+
+                assert checksum
+
+                return {"checksum": checksum, "path": path, "repo": repo}
+
+        except NotDvcRepoError:
+            repo_path = cached_clone(
+                self.def_repo[self.PARAM_URL],
+                rev=rev_lock or self.def_repo.get(self.PARAM_REV),
+            )
+            path = os.path.join(repo_path, self.def_path)
+            return {"checksum": None, "path": path, "repo": None}
+
+    @staticmethod
+    def _paths_changed(current_path, updated_path):
+        if not os.path.exists(current_path):
+            raise FileNotFoundError(current_path)
+
+        if not os.path.exists(updated_path):
+            raise FileNotFoundError(updated_path)
+
```

Comment:
Isn't `path` the same though? Except for location of a cloned repo, sure, but this is strange that you treat them as being completely different. Also, why not check existance and raise in _getfileinfo?

Also note that you can't really use existing paths when you take them out of `with _make_repo` contest. as it might be deleted on `__exit__`. It is not deleted now, but it is not nice to spread the context so much across your code.

Reply:
My reasoning for checking both is that the remote repository might not have it any more in the updated version. I checked the "past" one as a sanity check.

Reply:
Regarding the context, that one's going to be complicated to make readable :) I might have to open the contexts out in the `status` method so that they're still open and guaranteed to not be deleted.